src/toast/Toast.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {useLayoutEffect} from "react";
+
+import {useToaster} from "./util/toastHooks";
+import {ToastContextState} from "./util/toastTypes";
+import ListItem from "../list/item/ListItem";
+import {DEFAULT_TOAST_TIMEOUT} from "./util/toastConstants";
+
+export interface ToastProps {
+  testid: string;
+  data: ToastContextState["toastStack"][0];
+}
+
+function Toast({testid, data}: ToastProps) {
+  const {hide} = useToaster();
+  const {
+    autoClose = true,
+    timeout = DEFAULT_TOAST_TIMEOUT,
+    render,
+    customClassName,
+    id: toastId
+  } = data;
+
+  useLayoutEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+
+    if (autoClose) {
+      timeoutId = setTimeout(() => {
+        hide(toastId);
+      }, timeout);
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [autoClose, timeout, hide, toastId]);
+
+  return (
+    <ListItem testid={testid} customClassName={customClassName}>
+      {render()}
```

Comment:
We need to wrap with a context here so that any component defined under `render` can reach to this context to use, for example, this toast's id.

Reply:
Thank you for your advice 🙏🏻